include/version.h.in

```diff
@@ -1,5 +1,21 @@
 #pragma once
 
+#cmakedefine01 CELERITY_SYCL_IS_ACPP
+#cmakedefine01 CELERITY_SYCL_IS_DPCPP
+#cmakedefine01 CELERITY_SYCL_IS_SIMSYCL
+
+// CELERITY_DETAIL_ENABLE_DEBUG is specified on the command line
```

Comment:
Why is that? Is it so that it can be changed while keeping the same Cmake build?

Reply:
It's defined in terms of a generator expression to support multi-configuration builds (aka, switching between Debug and Release in MSVC).